<?xml version="1.0" encoding="UTF-8"?>

<launch>
    <!-- <arg name="initial_x" default="0" />
    <arg name="initial_y" default="0" />
    <arg name="des_x" default="-3.0" />
    <arg name="des_y" default="9.5" /> -->
    <group ns="/bat_swarm">
        <!-- <param name="des_pos_x" value="4.0" />
        <param name="des_pos_y" value="-8.0" /> -->
        <param name="des_pos_x" value="0.0" />
        <param name="des_pos_y" value="0.0" />
        <param name="Qmax" value="0.20"/>
        <param name="Qmin" value="0.06"/>
        <param name="NP" value="6"/>
        <param name="N_Gen" value="10000"/>
        <param name="D" value="2"/>
        <param name="Lower" value="-10"/>
        <param name="Upper" value="10"/>
        <param name="alpha" value="0.1"/>
        <!-- <param name="r_init" value="0.2"/> -->
        <param name="gamma" value="0.05"/>
        <node pkg="bat_algo" type="best_bat.py" name="best_bat" output="screen" />
        <node pkg="bat_algo" type="gen_checker.py" name="gen_checker" output = "screen" />
        <group ns="bat_1">
            <param name="initial_x" value="-5" />
            <param name="initial_y" value="-7" />
            <param name="A" value="0.69"/>
            <param name="r" value="0.2"/>
            <param name="robot_name" value="bat_1"/>
            <node pkg="bat_algo" type="reading_laser.py" name="reading_laser_1" output="screen" />
        </group>
        <group ns="bat_2">
            <param name="initial_x" value="-5.0" />
            <param name="initial_y" value="-8.0" />
            <param name="A" value="0.85"/>
            <param name="r" value="0.2"/>
            <param name="robot_name" value="bat_2"/>
            <node pkg="bat_algo" type="reading_laser.py" name="reading_laser_2" output="screen" />
        </group>
        <group ns="bat_3">
            <param name="initial_x" value="-3.0" />
            <param name="initial_y" value="-7.0" />
            <param name="A" value="0.78"/>
            <param name="r" value="0.2"/>
            <param name="robot_name" value="bat_3"/>
            <node pkg="bat_algo" type="reading_laser.py" name="reading_laser_3" output="screen" />
        </group>
        <group ns="bat_4">
            <param name="initial_x" value="-3.0" />
            <param name="initial_y" value="-8.0" />
            <param name="A" value="0.89"/>
            <param name="r" value="0.2"/>
            <param name="robot_name" value="bat_4"/>
            <node pkg="bat_algo" type="reading_laser.py" name="reading_laser_4" output="screen" />
        </group>
        <group ns="bat_5">
            <param name="initial_x" value="-2.0" />
            <param name="initial_y" value="-8.0" />
            <param name="A" value="0.78"/>
            <param name="r" value="0.2"/>
            <param name="robot_name" value="bat_5"/>
            <node pkg="bat_algo" type="reading_laser.py" name="reading_laser_5" output="screen" />
        </group>
        <group ns="bat_6">
            <param name="initial_x" value="-1.0" />
            <param name="initial_y" value="-9.0" />
            <param name="A" value="0.88"/>
            <param name="r" value="0.2"/>
            <param name="robot_name" value="bat_6"/>
            <node pkg="bat_algo" type="reading_laser.py" name="reading_laser_6" output="screen" />
        </group>
        <!-- <group ns="bat_7">
            <param name="initial_x" value="-4.0" />
            <param name="initial_y" value="-5.0" />
            <param name="A" value="0.70"/>
            <param name="r" value="0.2"/>
            <param name="robot_name" value="bat_7"/>
            <node pkg="bat_algo" type="reading_laser.py" name="reading_laser_7" output="screen" />
        </group>
        <group ns="bat_8">
            <param name="initial_x" value="5.0" />
            <param name="initial_y" value="8.0" />
            <param name="A" value="0.85"/>
            <param name="r" value="0.2"/>
            <param name="robot_name" value="bat_8"/>
            <node pkg="bat_algo" type="reading_laser.py" name="reading_laser_8" output="screen" />
        </group>
        <group ns="bat_9">
            <param name="initial_x" value="-2.0" />
            <param name="initial_y" value="5.0" />
            <param name="A" value="0.93"/>
            <param name="r" value="0.2"/>
            <param name="robot_name" value="bat_9"/>
            <node pkg="bat_algo" type="reading_laser.py" name="reading_laser_9" output="screen" />
        </group>
        <group ns="bat_10">
            <param name="initial_x" value="8.0" />
            <param name="initial_y" value="1.0" />
            <param name="A" value="0.98"/>
            <param name="r" value="0.2"/>
            <param name="robot_name" value="bat_10"/>
            <node pkg="bat_algo" type="reading_laser.py" name="reading_laser_10" output="screen" />
        </group> -->
    </group>
    <!-- <include file="$(find bat_algo)/launch/go_to_point.launch" />
    <include file="$(find bat_algo)/launch/move_bat.launch" /> -->
</launch>